package serverland;

message TranslationRequestMessage {
  required string request_id = 1;
  required string source_language = 2;
  required string target_language = 3;
  required string source_text = 4;
  optional string target_text = 5;

  message KeyValuePair {
    required string key = 1;
    required string value = 2;
  }

  repeated KeyValuePair packet_data = 6; // Container for additional request data.
}